openrelik_worker_common/utils.py

```diff
@@ -166,48 +142,60 @@ def to_dict(self):
             A dictionary containing the attributes of the OutputFile object.
         """
         return {
+            "uuid": self.uuid,
             "display_name": self.display_name,
+            "extension": self.extension,
             "data_type": self.data_type,
-            "uuid": self.uuid,
             "path": self.path,
             "original_path": self.original_path,
             "source_file_id": self.source_file_id,
         }
 
 
 def create_output_file(
-    output_path: str,
-    filename: Optional[str] = None,
-    uuid_as_filename: Optional[bool] = False,
-    add_extension: Optional[str] = None,
-    data_type: Optional[str] = "openrelik:worker:file:generic",
+    output_base_path: str,
+    display_name: Optional[str] = None,
+    extension: Optional[str] = None,
+    data_type: Optional[str] = None,
     original_path: Optional[str] = None,
     source_file_id: Optional[OutputFile] = None,
 ) -> OutputFile:
     """Creates and returns an OutputFile object.
 
     Args:
-        output_path: The path to the output directory.
-        filename: The name of the output file (optional).
-        uuid_as_filename: Use UUID as filename (optional).
-        add_extension: File extension (optional).
+        output_base_path: The path to the output directory.
+        display_name: The name of the output file (optional).
+        extension: File extension (optional).
         data_type: The data type of the output file (optional).
         original_path: The orignal path of the file (optional).
         source_file_id: The OutputFile this file belongs to (optional).
 
     Returns:
         An OutputFile object.
-
-        Raises:
-            ValueError: If both filename and uuid_as_filename arguments are provided.
     """
-    if filename and uuid_as_filename:
-        raise ValueError("Cannot provide both filename and uuid_as_filename arguments.")
+    # Create a new UUID for the file to use as filename on disk.
+    uuid = uuid4().hex
+
+    # If display_name is missing, set the file's UUID as display_name.
+    display_name = display_name if display_name else uuid
+
+    # Allow for an explicit extension to be set.
+    if extension:
+        extension = extension.lstrip(".")
+        display_name = f"{display_name}{extension}"
```

Comment:
I would add some tests on the display_name and extension as I think you strip the extension on the previous line but create the display_name *without* a dot between display_name and extension here.
So that would create:
extension = ".txt"
after strip = "txt"
display_name = "filenametxt"